x-pack/plugins/cases/public/components/all_cases/multi_select_filter.tsx

```diff
@@ -118,74 +123,87 @@ export const MultiSelectFilter = <T extends string, K extends string = string>({
   };
 
   return (
-    <EuiPopover
-      ownFocus
-      button={
-        <EuiFilterButton
-          data-test-subj={`options-filter-popover-button-${id}`}
-          iconType={buttonIconType || 'arrowDown'}
-          onClick={toggleIsPopoverOpen}
-          isSelected={isPopoverOpen}
-          numFilters={showActiveOptionsNumber ? options.length : undefined}
-          hasActiveFilters={showActiveOptionsNumber ? selectedOptionKeys.length > 0 : undefined}
-          numActiveFilters={showActiveOptionsNumber ? selectedOptionKeys.length : undefined}
-          aria-label={buttonLabel}
-        >
-          {buttonLabel}
-        </EuiFilterButton>
-      }
-      isOpen={isPopoverOpen}
-      closePopover={() => setIsPopoverOpen(false)}
-      panelPaddingSize="none"
-      repositionOnScroll
+    <EuiFilterGroup
+      css={css`
+        ${transparentBackground && 'background-color: transparent;'};
+      `}
     >
-      {isInvalid && (
-        <>
-          <EuiHorizontalRule margin="none" />
-          <EuiCallOut
-            title={limitReachedMessage}
-            color="warning"
-            size="s"
-            data-test-subj="maximum-length-warning"
-          />
-          <EuiHorizontalRule margin="none" />
-        </>
-      )}
-      <EuiSelectable<FilterOption<T, K>>
-        options={options}
-        searchable
-        searchProps={{
-          placeholder: buttonLabel,
-          compressed: false,
-          'data-test-subj': `${id}-search-input`,
-        }}
-        emptyMessage={i18n.EMPTY_FILTER_MESSAGE}
-        onChange={_onChange}
-        singleSelection={false}
-        renderOption={renderOption}
-      >
-        {(list, search) => (
-          <div
+      <EuiPopover
+        ownFocus
+        button={
+          <EuiFilterButton
             css={css`
-              width: 400px;
+              max-width: 186px;
             `}
+            data-test-subj={`options-filter-popover-button-${id}`}
+            iconType={buttonIconType || 'arrowDown'}
+            onClick={toggleIsPopoverOpen}
+            isSelected={isPopoverOpen}
+            numFilters={showActiveOptionsNumber ? options.length : undefined}
+            hasActiveFilters={showActiveOptionsNumber ? selectedOptionKeys.length > 0 : undefined}
+            numActiveFilters={showActiveOptionsNumber ? selectedOptionKeys.length : undefined}
+            aria-label={buttonLabel}
           >
-            <EuiPopoverTitle paddingSize="s">{search}</EuiPopoverTitle>
+            <EuiTextTruncate text={buttonLabel || ''} truncation={'end'} width={128}>
+              {(truncatedText: string) => (
+                <EuiText size="s" dangerouslySetInnerHTML={{ __html: truncatedText }} />
```

Comment:
do we need to sanitise this? I'd say no because it's the label we are already saving in ES so I would assume we are already sanitising it. What do you think?